iroha/src/smartcontracts/mod.rs

```diff
@@ -12,6 +12,9 @@ pub use isi::*;
 use super::wsv::WorldStateView;
 use crate::wsv::WorldTrait;
 
+/// Operation both mutable and immutable that has access to `WSV`.
+pub trait Operation {}
```

Comment:
Operation seems like a strange name

Reply:
What would you suggest?

Reply:
I would suggest to use `AsWSVMut` and `AsWSVRef` with analogy of `AsMut` and `AsRef`. May be not that kind of name, but i would think of that direction